### PR TITLE
Issue 58 nlp expansion

### PR DESCRIPTION
**Pull request by shayansiddiqui**

### CODE REVIEW

**Review by jyotirmay123 — CHANGES_REQUESTED** [review]
Lots of console.logs, could you please remove them or do you need them for further checking?

File: server/config.js [hunk 1]
```diff
@@ -9,21 +9,21 @@ let config = (function() {
...
+        this.nlp_model_sentence = 'model_20180407-125051',
+        this.nlp_model_ops = 'model_20180407-234649',
+        this.config_table = 'config',
 
         this.mysql = {
             host: 'localhost',
             port: '3306',
             user: 'root',
-            password: '',
+            password: 'root',
```
Comment (csehlke) [inline server/config.js]: you might want to remove your password again 😄

File: server/utils/dictionary.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    "function_alias":{
+        "count": "No of ",
+        "max": "Maximum ",
+        "min": "Minimum ",
+        "avg": "Average ",
+    },
+    "core_dataset":{
+        "age": ["employee’s age", "age of employees", "employee age", "distribution of age"],
+        "sex": ["employee’s sex", "sex of employees", "employee sex", "gender"],
+        "employment_status": ["status", "employee’s sex", "sex of employees", "employee status"],
```
Comment (csehlke) [inline server/utils/dictionary.js]: are you sure that this should be `employes sex` and not status?

Comment (csehlke) [reply server/utils/dictionary.js]: and maybe we should add the three phrases also for `gender`

**Review by csehlke — CHANGES_REQUESTED** [review]

**Review by csehlke — CHANGES_REQUESTED** [review]
Do we need the folders `projects/my_project` and `projects/test_projects` ?

File: UI/src/app/home/home.component.html [hunk 3]
```diff
@@ -1,88 +1,116 @@
...
+<hr>
+<div>
+    <div class="display-container">
+        <div id="error" [hidden]="!err_" class="msg-div">
+            <i class="fa fa-exclamation-triangle" style="font-size: 56px"></i>
+            <br>
+            <p>OOPS! Something went wrong.</p>
+        </div>
+        <div class="plot-div" [hidden]="err_">
+            <div [hidden]="!disableUpdateTextField()" class="msg-div">
```
Comment (csehlke) [inline UI/src/app/home/home.component.html]: you are abusing those methodes: 
I'd like to have the two text fields to be shown next to each other the whole time, like it was before

other than that the design looks really nice

**Review by csehlke — CHANGES_REQUESTED** [review]

**Review by csehlke — COMMENTED** [review]
where did you get the logo from? do we need to credit the author somewhere?

**Review by jyotirmay123 — APPROVED** [review]

File: UI/src/app/home/home.component.html (hunk 3 above)
Comment (csehlke) [reply UI/src/app/home/home.component.html]: I fixed it in #81 
maybe you take a look at it and merge it in here

**Comment by csehlke** [comment]
@shayansiddiqui can you please take a look the comments, so that we could merge this?
I will take care of the merge conflict due to my UI testing commits later this evening

File: UI/src/app/home/home.component.html (hunk 3 above)
Comment (shayansiddiqui) [reply UI/src/app/home/home.component.html]: As I understood you were using two diff functions for the same task(showing and hiding of text input and button) thats why I used it. Also the button disableUpdateButton() was not working for me so I used it everywhere. However you are free to experiment with layout!!

**Comment by shayansiddiqui** [comment]
Can somebody review and approve again?

**Review by jyotirmay123 — APPROVED** [review]
looks good.

**Review by iraklitchedia — APPROVED** [review]
Looks fine

**Review by csehlke — APPROVED** [review]

**Pull request merged**